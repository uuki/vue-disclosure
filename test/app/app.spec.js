import { mount, createLocalVue } from '@vue/test-utils'
import App from './test'
import VueDisclosure from '@/index.js'

describe('App.vue', () => {
  let wrapper;

  beforeEach(() => {
    const localVue = createLocalVue()
    localVue.use(VueDisclosure);

    wrapper = mount(App, { localVue });
  });

  it('adds an $disclosure scope to the Vue', () => {
    expect(typeof wrapper.vm.$disclosure).toBe('object')
    expect(typeof wrapper.vm.$disclosure.$window).toBeTruthy()
    expect(typeof wrapper.vm.$disclosure.active).toBeTruthy()
    expect(typeof wrapper.vm.$disclosure.setup).toBeTruthy()
    expect(typeof wrapper.vm.$disclosure.items).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(wrapper.element).toBeDefined();
    expect(wrapper.element).toMatchSnapshot();
  })

  it('test toggler component', () => {
    expect(wrapper.classes()).toContain('test')
    const toggler = wrapper.find('.togglerBasic')
    const content = wrapper.find('.contentBasic')

    expect(content.html().match(/style="([^"]*)"/i)[1]).toBe('max-height: 0px;')

    toggler.trigger('click')
    expect(content.classes()).toContain('vue-disclosure-active')
  })

})