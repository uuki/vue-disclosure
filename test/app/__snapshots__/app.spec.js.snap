// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should match snapshot 1`] = `
<div
  class="test"
>
  <h1>
    TEST
  </h1>
   
  <section
    class="pattern"
  >
    <h2
      class="pattern__head"
    >
      Basic
    </h2>
     
    <div
      class="pattern__content"
    >
      <div
        class="vue-disclosure-toggler togglerBasic"
      >
        Click Here
      </div>
       
      <div
        class="contentBasic vue-disclosure-content"
        style="max-height: 0px;"
      >
        Hello
      </div>
    </div>
  </section>
   
  <section
    class="pattern"
  >
    <h2
      class="pattern__head"
    >
      Responsive (demo 768px)
    </h2>
     
    <div
      class="pattern__content"
    >
      <div
        class="vue-disclosure-toggler"
      >
        Any Title
      </div>
       
      <div
        class="vue-disclosure-content"
        style="max-height: 0px;"
      >
        Any Contents
      </div>
    </div>
  </section>
</div>
`;
